A14-channelz.md

```diff
@@ -0,0 +1,1051 @@
+    of connections and RPCs, via its channels.  A client has channels, which it
+    uses to send RPCs. This term is used less commonly.
+
+
+# Channelz Data
+
+The channelz service exposes internal implementation details in a public way.
+Since different implementations have different internal details, the channelz
+data is lightly abstracted.  There are four top level entities exposed:
+Channels, Subchannels, Servers, and Sockets.
+
+Each entity is uniquely identified by a positive integer, called the id.  This
+id must be distinct and may not be reused over the lifetime of the process for
+any entity.  This ensures that querying channelz will never result in referring
+to the wrong entity.  While ids are not allocated sequentially, they are
+ordered.  This enables pagination of the results when querying channelz.
+Since protobuf treats the number 0 as the default value, it is unused as an id.
+
+Along with the id, a human readable name can optionally be associated with each
+the id.  The id and the name together form a "reference".  Thus, Channels,
+Servers and Sockets are identified by their respective references.  These
+references are abbreviated ChannelRefs, SubchannelRefs, ServerRefs, and 
+SocketRefs.  Note that only the id is necessary to query channelz.
+
+The data representation of each ref:
+
+![message ref structure][message ref structure]
+
+
+## Channels and Subchannels
+
+Channels and Subchannels, or descendent channel, are hierarchically organized
+into a DAG structure.  The union of all channels and subchannels may not contain
+a cycle.  A descendent channel may have any number of descendent channels. Each
+descendent channel may also have any number of sockets.   However, a given 
+descendent channel cannot have heterogeneous children.  That is, a channel or
+subchannel may have descendent channels, or have sockets, but not both.
+
+A subchannel represents a channel that is load-balanced over.  When a channel
+has both subchannels and channels, the channels are not delegated to by the 
+superchannel.  An example would be a channel that is used by the ancestor 
+channel, but does not handle RPCs given by the application.
+
+Each channel has a ChannelRef reference which includes the channel id and an
+optional name.  The name is included for human consumption.  There are no
+restrictions on the name but it should be limited to a reasonable length.
+
+Each subchannel has a SubchannelRef reference which includes the subchannel id
+and an optional name.  The name is included for human consumption.  There are 
+no restrictions on the name but it should be limited to a reasonable length.
+
+![channel hierarchy 1][channel hierarchy 1]
+
+### Channel Data
+
+Channels and Subchannels include data about themselves, notably their initial
+parameters and their call stats.  Currently the data includes:
+
+* Channel state ( See: connectivity-semantics-and-api.md )
```

Comment:
I assume this should be a link to https://github.com/grpc/grpc/blob/master/doc/connectivity-semantics-and-api.md.